annotation-processor/src/main/java/com/hubspot/annotation/processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,151 @@
+ */
+
+package com.hubspot.annotation.processor;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Processor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.tools.FileObject;
+import javax.tools.StandardLocation;
+
+import com.google.auto.service.AutoService;
+
+@SupportedAnnotationTypes("java.lang.SuppressWarnings")
+@AutoService(Processor.class)
+public class ModernizerAnnotationProcessor extends AbstractProcessor {
+
+    @Override
+    public final SourceVersion getSupportedSourceVersion() {
+        return SourceVersion.latest();
+    }
+
+    @Override
+    public final boolean process(
+        Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        for (TypeElement annotation : annotations) {
+            List<String> annotatedClasses =
+                getAnnotatedClasses(roundEnv, annotation);
+            makeAnnotatedClassesFile(annotatedClasses);
+        }
+        return true;
+    }
+
+    private List<String> getAnnotatedClasses(
+        RoundEnvironment roundEnv, TypeElement annotation) {
+        List<String> annotatedClasses = new ArrayList<String>();
+        for (Element element : roundEnv.getElementsAnnotatedWith(annotation)) {
+            List<String> warnings = Arrays.asList(
+                    element.getAnnotation(SuppressWarnings.class).value());
+            if (warnings.contains("modernizer")) {
+                if (element.getKind().isClass()) {
+                    annotatedClasses.add(getClassHeader(element));
+                }
+            }
+        }
+        return annotatedClasses;
+    }
+
+    private void makeAnnotatedClassesFile(List<String> annotatedClasses) {
+        FileWriter writer = null;
+        try {
+            File dir = new File(getFilePath(), "modernizer");
+            if (!dir.exists()) {
+                dir.mkdir();
+            }
+            File file = new File(dir.getPath(),
+                "modernizer-ignore-annotated-classes.txt");
+            writer = new FileWriter(file);
+            for (String element : annotatedClasses) {
+                writer.write(element + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (writer != null) {
+                try {
+                    writer.close();
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+    }
+
+    private String getFilePath() {
```

Comment:
I think it would make sense to call this `getOutputDirectory` and return a `File` object that already has `modernizer on the end of it`